### PR TITLE
Added Beit model ouput class

### PR DESCRIPTION
**Pull request by lumliolum**
# What does this PR do?

Fixes https://discuss.huggingface.co/t/issue-in-the-documentation-of-transformers-for-biet/10977/2

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

@NielsRogge

### CODE REVIEW

**Review by LysandreJik — APPROVED** [review]
Looks good to me! cc @sgugger

File: src/transformers/models/beit/modeling_beit.py [hunk 1]
```diff
@@ -86,6 +87,31 @@ def extra_repr(self) -> str:
         return "p={}".format(self.drop_prob)
 
 
+@dataclass
+class BeitModelOutputWithPooling(BaseModelOutputWithPooling):
```
Comment (sgugger) [inline src/transformers/models/beit/modeling_beit.py:91]: This should be added to the BeiT model doc page under a new section called "BeiT specific outputs". See the XLNet model doc source page for an example.

**Review by sgugger — APPROVED** [review]
Thanks a lot for adding this!

File: src/transformers/models/beit/modeling_beit.py (hunk 1 above)
Comment (lumliolum) [reply src/transformers/models/beit/modeling_beit.py:91]: I looked at this for reference : https://github.com/huggingface/transformers/blob/master/docs/source/model_doc/xlnet.rst#xlnet-specific-outputs

and added the `Specific Output` section.

File: docs/source/model_doc/beit.rst [hunk 2]
```diff
@@ -70,6 +70,13 @@ BeitConfig
     :members:
 
 
+XLNet specific outputs
```
Comment (sgugger) [inline docs/source/model_doc/beit.rst:73]:
```suggestion
BeiT specific outputs
```

Comment (NielsRogge) [reply docs/source/model_doc/beit.rst:73]: It's actually BEiT, not BeiT (nit)

Comment (sgugger) [reply docs/source/model_doc/beit.rst:73]: Edited, thanks for catching!

**Review by NielsRogge — APPROVED** [review]
LGTM! Can you just fix the `build_doc` issue on CI?

**Comment by lumliolum** [comment]
Hello,

I ran `make html SPHINXOPTS="-W -j 4"` in `docs` directory. This is the error I am getting

```
.virtualenvs/transformers_dev/lib/python3.7/site-packages/recommonmark/parser.py:75: UserWarning: Container node skipped: type=document
  warn("Container node skipped: type={0}".format(mdnode.t))

Warning, treated as an error:
missing attribute forward in object transformers.TapasModel
Makefile:19: recipe for target 'html' failed
make: *** [html] Error 2
```

Note: I ran the above command because in `Circle CI` I saw this
image

**Comment by NielsRogge** [comment]
I've opened a PR on your branch [here](https://github.com/lumliolum/transformers/pull/1).

**Comment by NielsRogge** [comment]
Thanks for adding this!

**Pull request merged**